modules/gltf/doc_classes/ModelDocument3D.xml

```diff
@@ -0,0 +1,59 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<class name="ModelDocument3D" inherits="Resource" experimental="" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="../../../doc/class.xsd">
+	<brief_description>
+	</brief_description>
+	<description>
+	</description>
+	<tutorials>
+	</tutorials>
+	<methods>
+		<method name="append_data_from_buffer">
+			<return type="int" enum="Error" />
+			<param index="0" name="bytes" type="PackedByteArray" />
+			<param index="1" name="base_path" type="String" />
+			<param index="2" name="state" type="ModelState3D" />
+			<param index="3" name="flags" type="int" default="0" />
+			<description>
+			</description>
+		</method>
+		<method name="append_data_from_file">
+			<return type="int" enum="Error" />
+			<param index="0" name="path" type="String" />
+			<param index="1" name="state" type="ModelState3D" />
+			<param index="2" name="flags" type="int" default="0" />
+			<param index="3" name="base_path" type="String" default="&quot;&quot;" />
+			<description>
+			</description>
+		</method>
+		<method name="append_data_from_scene">
+			<return type="int" enum="Error" />
+			<param index="0" name="node" type="Node" />
+			<param index="1" name="state" type="ModelState3D" />
+			<param index="2" name="flags" type="int" default="0" />
+			<description>
+			</description>
+		</method>
+		<method name="create_buffer">
+			<return type="PackedByteArray" />
+			<param index="0" name="state" type="ModelState3D" />
+			<description>
+			</description>
+		</method>
+		<method name="create_scene">
+			<return type="Node" />
```

Comment:
It needs to be a new name, I am not attached to `create_scene` but it cannot be `generate_scene`.

Reply:
Could be `create_scene_from_data` to match the pattern.

Reply:
`virtual Node *generate_scene_from_data(Ref<ModelState3D> p_state, float p_bake_fps = 30.0f, bool p_trimming = false, bool p_remove_immutable_tracks = true) override;`

vs

`Node *GLTFDocument::generate_scene_81746(Ref<GLTFState> p_state, float p_bake_fps, bool p_trimming, bool p_remove_immutable_tracks) {
`

Note `GLTFState` extends `ModelState3D`

Reply:
If we have to change it, I prefer create_scene like you had it.

but I am not satisfied with the answer you gave. Why can't we call it generate_scene given the existence of generate_scene_81746

Reply:
We can call it generate_scene but it will be expose to gdscript as `generate_scene_from_data` since it's a pattern in other parts of godot engine that we change name rather than change the parameter types.

Reply:
But we haven't changed any of the parameter types, have we?

Reply:
Ref<GLTFState> p_state became Ref<ModelState3D> p_state

Reply:
Two things:
1. A Ref<ModelState3D> parameter can cleanly accept a Ref<GLTFState> with no conversion loss. Therefore this change should not be compatiblity breaking, certainly not from the perspective of GDScript.

As for GDExtension, the bind_compat_method should take care of the compatibility.

2. FBXState is a subclass of GLTFState, so there is no need to change this argument type yet. We can delay using the correct ModelState3D type until whatever support for contravariance is missing from point 1 is implemented.

in conclusion, I still see no reason to require a new method name.

And I am fairly sure the same applies to the other methods such as append_data_...

Reply:
Applied. Please check.